pilot/pkg/dns/dns.go

```diff
@@ -44,6 +44,8 @@ type LocalDNSServer struct {
 	// Optimizations to save space and time
 	proxyDomain      string
 	proxyDomainParts []string
+	// Stores if lookupTable is updated atleast once.
+	initialized bool
```

Comment:
We could also check `lookupTable != nil` maybe?

Reply:
We have to use h.lookupTable.Load() != nil - was trying to avoid Load call. But may be OK. Changed now. PTAL